src/Interpreters/ProcessList.cpp

```diff
@@ -757,4 +767,69 @@ ProcessList::QueryAmount ProcessList::getQueryKindAmount(const IAST::QueryKind &
     return found->second;
 }
 
+void ProcessList::increaseWaitingQueryAmount(const QueryStatusPtr & status)
+{
+    UInt64 limit = max_waiting_queries_amount.load();
+    UInt64 value = waiting_queries_amount.load();
```

Comment:
"Shouldn't `value` acquisition happen inside the `while` loop?

Reply:
cc @serxa Please take a look at the comment (though I accepted it before that). 
I think its a bug.

Reply:
it's okay. `compare_exchange_strong` updates the `value` if it fails